bin/commitments_deprecater.rb

```diff
@@ -0,0 +1,190 @@
+# frozen_string_literal: true
+
+require "cluster"
+require "services"
+require "deprecation_record"
+
+Services.mongo!
+
+# Takes (shared print) deprecation records, and tries to match them to commitments
+# so that those commitments can be deprecated.
+# Usage: CommitmentsDeprecater.new.run(<list_of_files>)
+
+class CommitmentsDeprecater
+  attr_reader :err, :log_path, :report
+
+  def initialize(verbose: false)
+    @verbose = verbose
+    @header_spec = ["organization", "ocn", "local_id", "deprecation_status"]
+  end
+
+  # Process a list of files.
+  def run(args)
+    args.each do |arg|
+      if File.file?(arg)
+        from_file(arg)
+      else
+        log "arg #{arg} not a file, skipping"
+      end
+    end
+  end
+
+  # Write log message to file, and STDERR if @verbose
+  def log(msg)
+    if @report.nil?
+      # To reduce the risk of accidental overwrites by a different instance,
+      # make the filename rather unique-ish.
+      iso_stamp = Time.now.strftime("%Y%m%d-%H%M%S")
+      rand_str  = SecureRandom.hex(8)
+      log_dir   = Settings.deprecation_report_path
+      FileUtils.mkdir_p(log_dir)
+      @log_path = "#{log_dir}/commitments_deprecater_#{iso_stamp}_#{rand_str}.log"
+      warn "Logging to #{@log_path}" if @verbose
+      @report = File.open(@log_path, "w")
+    end
+    @report.puts msg
+    warn msg if @verbose
+  end
+
+  # Read deprecation records from file and try to deprecate the matching commitments.
+  def from_file(path)
+    inf = File.open(path, "r")
+    log "Reading deprecation records from #{path}"
+    if !check_header(inf.gets.strip)
```

Comment:
This is similar to `Loader::FileLoader.load`; consider adding an option to require the header there (rather than skipping it if it is present?)

Reply:
I am strongly pro-headers for tabular data.